### PR TITLE
Explain that missing consents might be expired.

### PR DESCRIPTION
**Pull request by sigurdm**
Also validate consent Ids before showing them.

### CODE REVIEW

File: app/lib/account/consent_backend.dart [hunk 1]
```diff
@@ -37,8 +37,15 @@ class ConsentBackend {
 
   ConsentBackend(this._db);
 
+  /// Throws if [consentId] is malformed.
+  void validateConsent(String consentId) {
+    InvalidInputException.checkMatchPattern(
+        consentId, 'consentId', RegExp(r'^[a-zA-Z0-9]*$'));
```
Comment (jonasfj) [inline app/lib/account/consent_backend.dart]: or add `InvalidInputException.checkUlid(value, name)` I imagine we'll use ulid's else where too.

File: app/lib/account/consent_backend.dart [hunk 2]
```diff
@@ -37,8 +37,15 @@ class ConsentBackend {
 
   ConsentBackend(this._db);
 
+  /// Throws if [consentId] is malformed.
+  void validateConsent(String consentId) {
```
Comment (isoos) [inline app/lib/account/consent_backend.dart]: make this private

File: app/lib/shared/exceptions.dart [hunk 3]
```diff
@@ -78,11 +78,10 @@ class InvalidInputException extends ResponseException {
   }
 
   /// Throw [InvalidInputException] if [value] doesn't match [pattern].
-  static void checkMatchPattern(String value, String name, Pattern pattern) {
```
Comment (isoos) [inline app/lib/shared/exceptions.dart]: Please match the method name and param (e.g. `checkMatchRegExp` or use `Pattern` instead of `RegExp`).

Comment (sigurdm) [reply app/lib/shared/exceptions.dart]: Done

**Pull request merged**